inference-engine/Doxyfile.in

```diff
@@ -0,0 +1,2494 @@
+# Doxyfile 1.8.13
```

Comment:
We likely won't use doxygen as we have a separate documentation engine based on markdown in /docs (correct me if I'm wrong here). Should we remove this?

Reply:
I wasn't sure whether we would utilize it or not, so I kept it.

I guess if we need it again, we can always re-add it from upstream.

Happy to take this out for now.

Reply:
I would prefer to follow up and keep this PR minimal.

Reply:
As we discussed with Erik in private I want this to be as clean as possible on the first commit for a variety of reasons. If something is not addressed here, we'll have to create an issue about this. Let me have another pass.

Reply:
@eu9ene your feedback here is valuable and likely all worthwhile changes. 

But I do not understand the reservation about creating issues. Issues are cheap. They open. They get closed. They document a work item. I do not agree with the philosophy that as many things as possible should be packed into a single pull request (ostensibly for the sake of keeping the issue count low).

In fact, I would rather see more issues get created in a project than fewer issues, even if some are marked as duplicates, or closed as invalid etc. Good software engineering work is done incrementally. Issues help document that process.